lib/Model.js

```diff
@@ -420,7 +420,7 @@ Model.conditionCheck = async function(NewModel, key, options, next) {
       let rangeKeyName = schema.rangeKey.name;
       conditionReq.Key[rangeKeyName] = schema.rangeKey.toDynamo(key[rangeKeyName], undefined, key);
     }
-    processCondition(conditionReq, options, schema);
+    processCondition(conditionReq, options, NewModel);
```

Comment:
I just realized, should this be `await`'ed?

Reply:
this is a good catch.